docs(CONTRIBUTING): improve contributor onboarding guidance (#13980)
Esri/calcite-components: CONTRIBUTING.md

Contributors require additional permissions to gain access to the Chromatic test integration suite, where visual changes can be reviewed by the team prior to merging PRs.

> [!WARNING]

> > Pull requests must come from a cloned repo with a feature branch. **GitHub forks are not supported** since workflows triggered from forks cannot access repository secrets (e.g., Chromatic tokens), causing visual snapshot and CI checks to fail. See [GitHub documentation](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#security-hardening-with-open-source-workflows).

Reviewer: nitpick:

```suggestion
> **Pull requests must come from a cloned repo with a feature branch**. GitHub forks are not supported since workflows triggered from forks cannot access repository secrets (e.g., Chromatic tokens), causing visual snapshot and CI checks to fail. Refer to [GitHub's documentation on security guidelines](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#security-hardening-with-open-source-workflows).
```

## Getting a development environment set up

> You need [Node](https://nodejs.org/en) installed for development. If you do not already have Node, we recommend using [Mise](https://mise.jdx.dev/) (follow steps 1–3 from the [Getting Started page](https://mise.jdx.dev/getting-started.html)), which automatically uses the Node and npm versions defined in the [`engines`](https://docs.npmjs.com/cli/v11/configuring-npm/package-json#engines) field of [`package.json`](./package.json). If you use a different Node version manager, ensure it matches the major Node and npm versions specified in `engines`.

Reviewer: nitpick:

```suggestion
You need [Node](https://nodejs.org/en) installed for development. If you do not already have Node, it is recommend to use [Mise](https://mise.jdx.dev/). Follow steps 1–3 from the [Getting Started page](https://mise.jdx.dev/getting-started.html), which automatically uses the Node and npm versions defined in the [`engines`](https://docs.npmjs.com/cli/v11/configuring-npm/package-json#engines) field of [`package.json`](./package.json). If you use a different Node version manager, ensure it matches the major Node and npm versions specified in `engines`.
```

Reviewer 2: I think we could clarify that Mise installs/manages Node versions.

Also, I don't fully understand the switch of the first sentence. The original reads better to me 🤔

```suggestion
You need [Node](https://nodejs.org/en) installed for development. If you do not already have Node, it is recommended to use [Mise](https://mise.jdx.dev/) to install and manage Node versions. Follow steps 1–3 from [Mise's Getting Started page](https://mise.jdx.dev/getting-started.html), which will result in automatically using the Node and npm versions defined in the [`engines`](https://docs.npmjs.com/cli/v11/configuring-npm/package-json#engines) field of [`package.json`](./package.json). If you use a different Node version manager, ensure it matches the major Node and npm versions specified in `engines`.
```
